packages/apps/job-launcher/server/src/modules/manifest/manifest.service.spec.ts

```diff
@@ -0,0 +1,795 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { faker } from '@faker-js/faker';
+import { createMock } from '@golevelup/ts-jest';
+import { Encryption } from '@human-protocol/sdk';
+import { HttpStatus } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { ethers } from 'ethers';
+import { AuthConfigService } from '../../common/config/auth-config.service';
+import { CvatConfigService } from '../../common/config/cvat-config.service';
+import { PGPConfigService } from '../../common/config/pgp-config.service';
+import { Web3ConfigService } from '../../common/config/web3-config.service';
+import { ErrorJob } from '../../common/constants/errors';
+import {
+  JobCaptchaMode,
+  JobCaptchaRequestType,
+  JobCaptchaShapeType,
+  JobRequestType,
+} from '../../common/enums/job';
+import { ControlledError } from '../../common/errors/controlled';
+import { listObjectsInBucket } from '../../common/utils/storage';
+import { StorageService } from '../storage/storage.service';
+import { Web3Service } from '../web3/web3.service';
+import {
+  createJobAudinoDto,
+  createJobCaptchaDto,
+  createJobCvatDto,
+  getMockedProvider,
+  getMockedRegion,
+  mockAuthConfigService,
+  mockCvatConfigService,
+  mockWeb3ConfigService,
+} from './fixtures';
+import { FortuneManifestDto } from './manifest.dto';
+import { ManifestService } from './manifest.service';
+
+jest.mock('../../common/utils/storage', () => ({
+  ...jest.requireActual('../../common/utils/storage'),
+  listObjectsInBucket: jest.fn(),
+}));
+
+describe('ManifestService', () => {
+  let manifestService: ManifestService;
+  let storageService: StorageService;
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      providers: [
+        ManifestService,
+        { provide: Web3Service, useValue: createMock<Web3Service>() },
+        { provide: StorageService, useValue: createMock<StorageService>() },
+        {
+          provide: AuthConfigService,
+          useValue: mockAuthConfigService,
+        },
+        {
+          provide: CvatConfigService,
+          useValue: mockCvatConfigService,
+        },
+        { provide: PGPConfigService, useValue: { encrypt: false } },
+        {
+          provide: Web3ConfigService,
+          useValue: mockWeb3ConfigService,
+        },
+        { provide: Encryption, useValue: createMock<Encryption>() },
+      ],
+    }).compile();
+
+    manifestService = moduleRef.get<ManifestService>(ManifestService);
+    storageService = moduleRef.get<StorageService>(StorageService);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('createManifest', () => {
+    describe('createCvatManifest', () => {
+      let jobBounty: string;
+      beforeEach(() => {
+        jobBounty = faker.number.int({ min: 1, max: 1000 }).toString();
+        jest
```

Comment:
You don't need to create new spy before every test, it can be created once in `beforeAll` hook on the same `describe` block